### ISSUE TITLE
Improve DI documentation

### ISSUE THREAD

**Issue by danuvian** [activej/activej#101]
I found the documentation to the dependency injection section (https://inject.activej.io/docs.html) hard to read and understand. While it "explains" things and has examples, I honestly do not know what I am reading, coming from a Spring Boot background. I just want to be able to easily define a "component/bean" (in SB speak) and have it used elsewhere in the code. But the documentation talks about keys, bindings, types, recipes, modules, Trie, etc. Honestly, it does not make much sense to me, and some of it is implementation details, which could be put into another more advanced section and do not belong there.

I finally was able to get a simple DI example to work with something like this:

@Provides
String message() { return "Hello World"; }

@Inject
String message;

I googled a lot and finally found the above snippet. There is very little to no documentation and examples on the Internet on how to use activej's inject. Your documentation should have something like the above.

However, the snippet above only works within the same class file that contains the HttpServerLauncher. I have something like this:
public class WebApp extends HttpServerLauncher { 
    @Provides
    String message() { return "Hello World"; }
    
    @Inject
    String message;

    // code that uses the message variable in a REST call
}

However I could not find anywhere something like "component scanning" that is in Spring Boot. Basically I wanted to have a second file/class where I defined these "beans" and have it be picked up by the DI engine in activej. I have no idea how to "register" those "beans" in another class.

I think if you made the documentation and samples easier to read, remove a lot of the low-level details, and just gave us instructions on how to use DI quickly and easily, that would make the documentation a lot better. I don't think you should be telling us implementation details such as Trie and such.

Basically, add instructions for using @Provides and @Inject like my examples above, and show us how to do "component scanning" or it's equivalent. Show us the equivalent of SB's @Bean for creating beans for DI. The current documentation with the cookie examples, modules and recipes are, quite frankly, unhelpful and do not increase understanding.

Thanks in advance.

**Comment by gsitgithub** [comment]
Having same problem, an example may be useful like sample **MODULAR** application assuming N number of dependencies in N number of modules.
Great job btw.
Thanks

**Comment by eduard-vasinskyi** [comment]
Hello, @danuvian, @gsitgithub

[ActiveJ Inject](https://activej.io/inject) has a modern high-performance implementation of a module structure with concepts similar to the ones of the [Google Guice DI library](https://github.com/google/guice). All dependencies are defined in modules, either explicitly (with DSL) or implicitly, with annotations. The component scanning on module class/instance level is also supported, with the `ModuleBuilder.scan()` method. The modules are explicitly passed to the `Injector`. `Injector` combines those modules together, detects missing dependencies, cycles, etc. 

The example you provided (the one with `@Provides` and `@Inject` annotations) works as `@Inject` is placed inside the `Launcher` class. `Launcher` actually uses `Injector` to inject instances into itself. It is done via the `InstanceInjector` class and it is not an automatic process.

We are currently updating our website as well as documentation. I have added a [quick start](https://activej.io/inject/quick-start) tutorial that shows how to provide components in different modules and then use `Injector` to obtain a required instance. [Another tutorial](https://activej.io/inject/injecting-instances) describes injecting dependencies into arbitrary classes. There is also a new [class scanning tutorial](https://activej.io/inject/class-scanning) that demonstrates how to use `@Provides` annotation outside of modules. 

Please, let us know if you have any questions about ActiveJ Inject or its documentation. We are still in the process of updating documentation/examples, any suggestions on documentation improvements are welcomed.
